layers/+email/mu4e/packages.el

```diff
@@ -38,4 +39,10 @@
 
       (when mu4e-account-alist
         (add-hook 'mu4e-compose-pre-hook 'mu4e/set-account)
-        (add-hook 'message-sent-hook 'mu4e/mail-account-reset)))))
+        (add-hook 'message-sent-hook 'mu4e/mail-account-reset))
+      (if mu4e-use-maildirs
+        (mu4e-maildirs-extension)))))
+
+(defun mu4e/init-mu4e-maildirs-extension ()
+  (use-package mu4e-maildirs-extension
+    :commands (mu4e-maildirs-extension)))
```

Comment:
So.. I was counting on the implicit :defer due to having a :commands. Are there situations where mu4e/init-mu4e-maidirs-extension is evaluated prior to the user setting the mu4-use-maildirs variable but before the mu4e :config is run?